### PR TITLE
Add FLEXIQuant-LF to data analysis steps

### PR DESCRIPTION
**Pull request by lucatreide**
## Description

Adds FLEXIQuant-LF as a data analysis step to PROTzilla.

This is part of Luca's Bachelor Thesis. 

## Testing

Test are going to be added later on.

### CODE REVIEW

File: protzilla/data_analysis/ptm_quantification.py [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+
+
+def flexiquant_lf(
+    peptide_df: pd.DataFrame,
+    metadata_df: pd.DataFrame,
+    reference_group: str,
+    protein_id: str,
+    num_init: int = 50,
+    mod_cutoff: float = 0.5,
+) -> dict:
```
Comment (JanniRoebbecke) [inline protzilla/data_analysis/ptm_quantification.py:27]: Please add or complete docstrings for all the new Methods here (with description of the generell behavior, the parameters and the return value) (in standard PROTzilla format)

File: protzilla/data_analysis/ptm_quantification.py [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+
+    df = pd.merge(
+        left=df,
+        right=metadata_df[["Sample", "Group"]],
+        on="Sample",
+        copy=False,
+    )
+
+    if not "Group" in df:
+        return dict(
```
Comment (JanniRoebbecke) [inline protzilla/data_analysis/ptm_quantification.py:54]: The chosen metadata column should probably not be hard coded, but chosen by the user.

File: protzilla/data_analysis/ptm_quantification.py [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+    matrix_distance_RL.loc[idx] = distance_RL
+
+    # add CBs as columns to dataframe_train
+    dataframe_train["CB low"] = CB_low
+    dataframe_train["CB high"] = CB_high
+
+    return matrix_distance_RL, dataframe_train
+
+
+def create_regression_plots(
```
Comment (JanniRoebbecke) [inline protzilla/data_analysis/ptm_quantification.py:408]: Screenshot 2024-06-25 at 10-00-25 PROTzilla - agg

The Points should be different colors

File: protzilla/importing/peptide_import.py [hunk 4]
```diff
@@ -33,25 +33,37 @@ def peptide_import(file_path, intensity_name, map_to_uniprot) -> dict:
...
+        ordered.sort_values(
+            by=["Sample", "Protein ID"], ignore_index=True, inplace=True
+        )
+    else:
+        final_df = df.rename(columns={"Proteins": "Protein ID"})
+        ordered = final_df[["Sample", "Protein ID", "Sequence", "Intensity", "PEP"]]
+        ordered.dropna(subset=["Protein ID"], inplace=True)
+        ordered.sort_values(
+            by=["Sample", "Protein ID"], ignore_index=True, inplace=True
+        )
```
Comment (JanniRoebbecke) [inline protzilla/importing/peptide_import.py]: duplicate code behavior in the two cases of the if

**Review by JanniRoebbecke — APPROVED** [review]

File: protzilla/data_analysis/ptm_quantification.py (hunk 3 above)
Comment (lucatreide) [reply protzilla/data_analysis/ptm_quantification.py:408]: Yes and no. The color is about the RM_score, so this is actually correct. Problem is that you cannot really use the legend, I will fix this by using plotly for that plot. (in a different pr)

**Pull request merged**